### ISSUE TITLE
Integration tests don’t run on device, only sim

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-gl-native#11297]
Attempting to run the Integration Test Harness tests on a physical device fails with:

> Test target integration encountered an error (Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980 If you believe this error represents a bug, please attach the log file…

> Error starting test manager service: Error Domain=XCTMobileDeviceFramework Code=101 “kAMDMuxConnectError: Could not connect to the device.” UserInfo={NSLocalizedDescription=kAMDMuxConnectError: Could not connect to the device.}

... or:

> Test target integration encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted)

This is in: Xcode 9.2, iOS 11.2.x

/cc @akitchen @asheemmamoowala

**Comment by friedbunny** [comment]
I don’t see any good hints in the logs — Xcode is being cryptic about this one.

<details>
  <summary>Here’s the full test-run failure on Xcode 9.3b3. Strangely, it looks like it’s also trying to connect to my iPad wirelessly, even though I’ve told it to use my wired iPhone.</summary>

```
21:39:30.894 Xcode[37529:2561629] Beginning test session integration-287E09B7-2E67-4254-81C3-C4C9D5A503ED at 2018-02-23 21:39:30.893 with Xcode 9Q117m on target <DVTiOSDevice: 0x7fa81a668e40> {
		deviceSerialNumber:         xxxxxxxxxxxx
		identifier:                 3adfed5433a71edb24353deab9a2cb6ded4e9980
		deviceClass:                iPhone
		deviceName:                 Hobbes X
		deviceIdentifier:           3adfed5433a71edb24353deab9a2cb6ded4e9980
		productVersion:             11.2.6
		buildVersion:               15D100
		deviceSoftwareVersion:      11.2.6 (15D100)
		deviceArchitecture:         arm64
		deviceTotalCapacity:        60335489024
		deviceAvailableCapacity:    17102303232
		deviceIsTransient:          NO
		ignored:                    NO
		deviceIsBusy:               NO
		deviceIsActivated:          YES
		deviceActivationState:      Activated
		isPasscodeLocked:           NO
		deviceType:                 <DVTDeviceType:0x7fa81a2a1480 Xcode.DeviceType.iPhone>
		supportedDeviceFamilies:    (
    1
)
		applications:              (null)
		provisioningProfiles:      (null)
		hasInternalSupport:        NO
		isSupportedOS:             YES
		developerDiskMountError:   (null)
(null)
	bootArgs:                  <unavailable>
		connected:                 yes
		isWirelessEnabled:         yes
		connectionType:            direct
		} (11.2.6 (15D100))
21:39:30.894 Xcode[37529:2561629] Launching with Xcode.DebuggerFoundation.Launcher.LLDB
21:39:30.898 Xcode[37529:2561629] Test standard output and standard error is at /var/folders/s5/2p4466vn2cv5mslx9ydthw6m0000gn/T/com.apple.dt.XCTest/IDETestRunSession-D1A04C61-F466-4BD4-8A45-0F0E5FF380F4/integration-EB0A988E-19C2-40F5-9C51-3A4B95A42D42/StandardOutputAndStandardError.txt
21:39:52.979 Xcode[37529:2561629] ################ HAS CONNECTED iPad  ###################
21:39:52.979 Xcode[37529:2561629] Set DTDeviceKitLogLevel to verbose (3) for detailed connection logs
21:39:53.817 Xcode[37529:2568492] ## Unable to mount developer disk image, (Error Domain=com.apple.dtdevicekit Code=-402653183 "An unknown error occurred." UserInfo={NSLocalizedDescription=An unknown error occurred., com.apple.dtdevicekit.stacktrace=(
	0   DTDeviceKitBase                     0x0000000119d9cab6 DTDKCreateNSError + 113
	1   DTDeviceKitBase                     0x0000000119d9d2fa DTDK_AMDErrorToNSError + 1135
	2   DTDeviceKitBase                     0x0000000119ddd418 -[DTDKMobileDeviceToken mountDeveloperDiskImage:withError:] + 774
	3   DTDeviceKitBase                     0x0000000119dddcc8 -[DTDKMobileDeviceToken mountDeveloperDiskImageWithError:] + 479
	4   IDEiOSSupportCore                   0x0000000119cba330 __37-[DVTiOSDevice(Connect) hasConnected]_block_invoke_2 + 133
	5   DVTFoundation                       0x000000010aaeb03d __DVTDispatchGroupAsync_block_invoke + 806
	6   libdispatch.dylib                   0x00007fff58b9864a _dispatch_call_block_and_release + 12
	7   libdispatch.dylib                   0x00007fff58b90e08 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x00007fff58ba3f50 _dispatch_continuation_pop + 599
	9   libdispatch.dylib                   0x00007fff58b9b783 _dispatch_async_redirect_invoke + 703
	10  libdispatch.dylib                   0x00007fff58b929f9 _dispatch_root_queue_drain + 515
	11  libdispatch.dylib                   0x00007fff58b927a5 _dispatch_worker_thread3 + 101
	12  libsystem_pthread.dylib             0x00007fff58ed7169 _pthread_wqthread + 1387
	13  libsystem_pthread.dylib             0x00007fff58ed6be9 start_wqthread + 13
), NSLocalizedRecoverySuggestion=An unknown error occurred., NSLocalizedFailureReason=An unknown error occurred.}) {
    NSLocalizedDescription = "An unknown error occurred.";
    NSLocalizedFailureReason = "An unknown error occurred.";
    NSLocalizedRecoverySuggestion = "An unknown error occurred.";
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x0000000119d9cab6 DTDKCreateNSError + 113
	1   DTDeviceKitBase                     0x0000000119d9d2fa DTDK_AMDErrorToNSError + 1135
	2   DTDeviceKitBase                     0x0000000119ddd418 -[DTDKMobileDeviceToken mountDeveloperDiskImage:withError:] + 774
	3   DTDeviceKitBase                     0x0000000119dddcc8 -[DTDKMobileDeviceToken mountDeveloperDiskImageWithError:] + 479
	4   IDEiOSSupportCore                   0x0000000119cba330 __37-[DVTiOSDevice(Connect) hasConnected]_block_invoke_2 + 133
	5   DVTFoundation                       0x000000010aaeb03d __DVTDispatchGroupAsync_block_invoke + 806
	6   libdispatch.dylib                   0x00007fff58b9864a _dispatch_call_block_and_release + 12
	7   libdispatch.dylib                   0x00007fff58b90e08 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x00007fff58ba3f50 _dispatch_continuation_pop + 599
	9   libdispatch.dylib                   0x00007fff58b9b783 _dispatch_async_redirect_invoke + 703
	10  libdispatch.dylib                   0x00007fff58b929f9 _dispatch_root_queue_drain + 515
	11  libdispatch.dylib                   0x00007fff58b927a5 _dispatch_worker_thread3 + 101
	12  libsystem_pthread.dylib             0x00007fff58ed7169 _pthread_wqthread + 1387
	13  libsystem_pthread.dylib             0x00007fff58ed6be9 start_wqthread + 13
);
}
21:39:53.818 Xcode[37529:2561629] ################ FINISHED HAS CONNECTED iPad (0) ################### (Error Domain=com.apple.dtdevicekit Code=-402653183 "An unknown error occurred." UserInfo={NSLocalizedDescription=An unknown error occurred., com.apple.dtdevicekit.stacktrace=(
	0   DTDeviceKitBase                     0x0000000119d9cab6 DTDKCreateNSError + 113
	1   DTDeviceKitBase                     0x0000000119d9d2fa DTDK_AMDErrorToNSError + 1135
	2   DTDeviceKitBase                     0x0000000119ddd418 -[DTDKMobileDeviceToken mountDeveloperDiskImage:withError:] + 774
	3   DTDeviceKitBase                     0x0000000119dddcc8 -[DTDKMobileDeviceToken mountDeveloperDiskImageWithError:] + 479
	4   IDEiOSSupportCore                   0x0000000119cba330 __37-[DVTiOSDevice(Connect) hasConnected]_block_invoke_2 + 133
	5   DVTFoundation                       0x000000010aaeb03d __DVTDispatchGroupAsync_block_invoke + 806
	6   libdispatch.dylib                   0x00007fff58b9864a _dispatch_call_block_and_release + 12
	7   libdispatch.dylib                   0x00007fff58b90e08 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x00007fff58ba3f50 _dispatch_continuation_pop + 599
	9   libdispatch.dylib                   0x00007fff58b9b783 _dispatch_async_redirect_invoke + 703
	10  libdispatch.dylib                   0x00007fff58b929f9 _dispatch_root_queue_drain + 515
	11  libdispatch.dylib                   0x00007fff58b927a5 _dispatch_worker_thread3 + 101
	12  libsystem_pthread.dylib             0x00007fff58ed7169 _pthread_wqthread + 1387
	13  libsystem_pthread.dylib             0x00007fff58ed6be9 start_wqthread + 13
), NSLocalizedRecoverySuggestion=An unknown error occurred., NSLocalizedFailureReason=An unknown error occurred.}), {
    NSLocalizedDescription = "An unknown error occurred.";
    NSLocalizedFailureReason = "An unknown error occurred.";
    NSLocalizedRecoverySuggestion = "An unknown error occurred.";
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x0000000119d9cab6 DTDKCreateNSError + 113
	1   DTDeviceKitBase                     0x0000000119d9d2fa DTDK_AMDErrorToNSError + 1135
	2   DTDeviceKitBase                     0x0000000119ddd418 -[DTDKMobileDeviceToken mountDeveloperDiskImage:withError:] + 774
	3   DTDeviceKitBase                     0x0000000119dddcc8 -[DTDKMobileDeviceToken mountDeveloperDiskImageWithError:] + 479
	4   IDEiOSSupportCore                   0x0000000119cba330 __37-[DVTiOSDevice(Connect) hasConnected]_block_invoke_2 + 133
	5   DVTFoundation                       0x000000010aaeb03d __DVTDispatchGroupAsync_block_invoke + 806
	6   libdispatch.dylib                   0x00007fff58b9864a _dispatch_call_block_and_release + 12
	7   libdispatch.dylib                   0x00007fff58b90e08 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x00007fff58ba3f50 _dispatch_continuation_pop + 599
	9   libdispatch.dylib                   0x00007fff58b9b783 _dispatch_async_redirect_invoke + 703
	10  libdispatch.dylib                   0x00007fff58b929f9 _dispatch_root_queue_drain + 515
	11  libdispatch.dylib                   0x00007fff58b927a5 _dispatch_worker_thread3 + 101
	12  libsystem_pthread.dylib             0x00007fff58ed7169 _pthread_wqthread + 1387
	13  libsystem_pthread.dylib             0x00007fff58ed6be9 start_wqthread + 13
);
}
21:43:08.555 Xcode[37529:2561629] ################ HAS CONNECTED iPad  ###################
21:43:08.555 Xcode[37529:2561629] Set DTDeviceKitLogLevel to verbose (3) for detailed connection logs
21:45:17.089 Xcode[37529:2561629] Launch session started.
21:45:17.190 Xcode[37529:2561629] Setting up test runner session
21:45:17.212 Xcode[37529:2585867] Connecting to mobile device: 3adfed5433a71edb24353deab9a2cb6ded4e9980
21:45:17.216 Xcode[37529:2585867] Starting session with mobile device
21:45:17.251 Xcode[37529:2585867] Starting test manager service on mobile device.
21:45:17.256 Xcode[37529:2585867] Error starting test manager service: Error Domain=XCTMobileDeviceFramework Code=101 "kAMDMuxConnectError: Could not connect to the device." UserInfo={NSLocalizedDescription=kAMDMuxConnectError: Could not connect to the device.}
21:45:17.258 Xcode[37529:2569640] Error returned from opening test runner connection: Error Domain=com.apple.dt.xctest.error Code=4 "Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980" UserInfo={NSLocalizedDescription=Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980, NSUnderlyingError=0x7fa8238108a0 {Error Domain=XCTMobileDeviceFramework Code=101 "kAMDMuxConnectError: Could not connect to the device." UserInfo={NSLocalizedDescription=kAMDMuxConnectError: Could not connect to the device.}}}
21:45:17.258 Xcode[37529:2561629] Failed to make test runner session:
Error Domain=com.apple.dt.xctest.error Code=4 "Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980" UserInfo={NSLocalizedDescription=Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980, NSUnderlyingError=0x7fa8238108a0 {Error Domain=XCTMobileDeviceFramework Code=101 "kAMDMuxConnectError: Could not connect to the device." UserInfo={NSLocalizedDescription=kAMDMuxConnectError: Could not connect to the device.}}}
21:45:17.259 Xcode[37529:2561629] Test operation failure: Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980
21:45:17.259 Xcode[37529:2561629] _finishWithError:Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980" UserInfo={NSLocalizedDescription=Unable to connect to test manager on 3adfed5433a71edb24353deab9a2cb6ded4e9980}
```

</details>

**Comment by julianrex** [comment]
Closing; believed fixed by #11563.
